### PR TITLE
[Rust] Upgrade to Rust version 1.74.1.

### PR DESCRIPTION
**Pull request by JoshLind**
### Description
This PR updates `aptos-core` to use rust version `1.74.1` (from `1.72.1`). Rust `1.74` has baked for > a [month](https://blog.rust-lang.org/2023/11/16/Rust-1.74.0.html). We're currently 3+ months behind.

### Test Plan
Existing test infrastructure.

### CODE REVIEW

File: crates/aptos-log-derive/src/lib.rs [hunk 1]
```diff
@@ -147,6 +147,7 @@ fn extract_internal_type(ty: &Type) -> Option<&Type> {
     None
 }
 
+#[allow(clippy::never_loop)]
```
Comment (gregnazario) [inline crates/aptos-log-derive/src/lib.rs:150]: What is this one?

File: execution/block-partitioner/src/v2/types.rs [hunk 2]
```diff
@@ -20,6 +20,7 @@ impl Ord for SubBlockIdx {
     }
 }
 
+#[allow(clippy::non_canonical_partial_ord_impl)]
```
Comment (gregnazario) [inline execution/block-partitioner/src/v2/types.rs:23]: And this one?

File: third_party/move/move-ir-compiler/move-ir-to-bytecode/syntax/src/syntax.rs [hunk 3]
```diff
@@ -1483,6 +1483,7 @@ fn parse_acquire_list(
     Ok(al)
 }
 
+#[allow(clippy::four_forward_slashes)]
 //// Spec language parsing ////
```
Comment (gregnazario) [inline third_party/move/move-ir-compiler/move-ir-to-bytecode/syntax/src/syntax.rs:1487]: Why not use 2?  A few of the other ones your converted from 4 to 3 might have been converted to unexpected doc comments

File: third_party/move/move-analyzer/src/symbols.rs [hunk 4]
```diff
@@ -1,6 +1,9 @@
 // Copyright (c) The Move Contributors
 // SPDX-License-Identifier: Apache-2.0
 
+#![allow(clippy::unwrap_or_default)]
+#![allow(clippy::non_canonical_partial_ord_impl)]
```
Comment (gregnazario) [inline third_party/move/move-analyzer/src/symbols.rs:5]: Any ones of note?

File: third_party/move/move-ir-compiler/move-ir-to-bytecode/syntax/src/syntax.rs (hunk 3 above)
Comment (JoshLind) [reply third_party/move/move-ir-compiler/move-ir-to-bytecode/syntax/src/syntax.rs:1487]: SGTM. 😄 I'll convert them to 2 -- kinda figured folks wanted docs comments (except in this particular case), but I'll use 2 and let them update.

File: crates/aptos-log-derive/src/lib.rs (hunk 1 above)
Comment (JoshLind) [reply crates/aptos-log-derive/src/lib.rs:150]: There's a weird loop pattern in this function (it's unclear to me what the author was trying to do). I've updated it according to the clippy (https://rust-lang.github.io/rust-clippy/master/index.html#/never_loop). I'm trying to minimize any potential breakages 😄

File: third_party/move/move-analyzer/src/symbols.rs (hunk 4 above)
Comment (JoshLind) [reply third_party/move/move-analyzer/src/symbols.rs:5]: Both of these are used many times throughout the file. Seemed cleaner to allow them for the entire file 😄

**Review by ibalajiarun — APPROVED** [review]
`consensus`, `docker`, `aptos-logger` changes LGTM.

**Review by banool — APPROVED** [review]
Mighty fine, looks like some efficiency wins too, eg filtering before cloning.

**Pull request merged**